app/Http/Controllers/API/TraderController.php

```diff
@@ -173,6 +179,20 @@ public function emailVoucherHistory(Request $request, Trader $trader)
         return response()->json(['message' => $response_text], 202);
     }
 
+    public function getMinMaxVoucherDates($vouchers) {
+        $sorted_vouchers = $vouchers->sortBy(function($voucher) {
+            return $voucher->paymentPendedOn->created_at->timestamp;
+        });
+
+        $max_date = $sorted_vouchers[0]->paymentPendedOn->created_at->format('d-m-Y');
```

Comment:
It turned out I had forgotten to run `->values()->all()` on the returned collection from the sort, before I did that the array elements were in an odd order.